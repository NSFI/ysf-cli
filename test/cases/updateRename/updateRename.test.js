const path = require("path");
const fsExtra = require("fs-extra");
const shell = require("shelljs");
const glob = require("glob");
describe("UpdateMethod", () => {
  beforeAll(() => {
    // 重建测试环境

    process.chdir(__dirname); // jest.mock path resolve
    shell.rm("-rf", "./testOutput");
    shell.cp("-R", "./reset", "./testOutput");

    jest.mock("../../../src/update/variables", () => {
      const path = require("path");
      return {
        PROJ_DIR: path.resolve("./testOutput"), //根目录
        CACHE_DIR: path.resolve("./")
      };
    });
  });
  it("should rename files", async () => {
    //  prepare
    const updateMethod = require("../../../src/update/updateMethods");

    const updateConfig = await fsExtra.readJSON(
      path.resolve(__dirname, "./u/update.json")
    );

    //  execute
    await updateMethod.rename(updateConfig);

    //  verify
    let expectedFiles = await glob.sync("testOutput/**", {
      cwd: __dirname,
      nodir: true
    });

    expect(expectedFiles).toHaveLength(5);
    expect(expectedFiles).toMatchSnapshot();
  });
  afterAll(() => {
    jest.resetAllMocks();
  });
});
