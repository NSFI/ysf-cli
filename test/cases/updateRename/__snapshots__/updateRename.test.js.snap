// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateMethod should rename files 1`] = `
Array [
  "testOutput/Afile",
  "testOutput/fileB",
  "testOutput/fileC",
  "testOutput/folderAfterRename/subFolder/files",
  "testOutput/newFolder/newFilename",
]
`;
